### PR TITLE
feat: Add SDK-wide delete method, cancel methods to Job and TrainingJob

### PR DESCRIPTION
**Pull request by vinnysenthil**
## Summary of changes
- Add `_delete_method` abstract property to `AiPlatformResourceNoun` class, requiring a resource noun to define it's delete method in GAPIC (i.e. `Dataset` would set this property to `delete_dataset`)
- Add abstract property `_cancel_method` to `_Job` class to force children to define it's cancel method (i.e. `BatchPredictionJob` would set `cancel_batch_prediction_job`
- Add high-level `delete()` method to `AiPlatformResourceNounWithFutureManager` that calls subclass deletion method
- Shifted the `optional_sync` decorator higher in `base.py` to be referenced by `delete()` for async capability
- Removed `Dataset.delete()` and `Model.delete()` but left their tests; they now test the high-level delete method
- Add `_TrainingJob.cancel()` and `_Job.cancel()` along with unit tests

Fixes [b/179510193](http://b/179510193) and [b/179945882](http://b/179945882) 🦕

### CODE REVIEW

File: google/cloud/aiplatform/models.py [hunk 1]
```diff
@@ -950,7 +951,7 @@ def delete(self, force: bool = False, sync: bool = True) -> None:
         if force:
             self.undeploy_all(sync=sync)
 
-        self._delete(sync=sync)
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/models.py:953]: Can this method be removed?

**Review by sasha-gitg — APPROVED** [review]
LGTM

File: google/cloud/aiplatform/models.py (hunk 1 above)
Comment (vinnysenthil) [reply google/cloud/aiplatform/models.py:953]: Yup, good call. Dropping it

**Review by ivanmkc — APPROVED** [review]
LGTM

**Pull request merged**